File: benchmarks/node/file_io.js
const fs = require("fs");
const path = "tn_benchmark_file_io.tmp";
const chunk = Buffer.from("TezzNativeBenchmarkBlock");
const reps = 12000;
const expected = chunk.length * reps;

const out = fs.openSync(path, "w");
for (let i = 0; i < reps; i += 1) {
  fs.writeSync(out, chunk);
}
fs.closeSync(out);

if (fs.statSync(path).size !== expected) {
  fs.rmSync(path, { force: true });
  process.exit(1);
}

const input = fs.openSync(path, "r");
const buf = Buffer.alloc(chunk.length);
let total = 0;
let acc = 0;
while (total < expected) {
  const got = fs.readSync(input, buf, 0, chunk.length, null);
  if (got !== chunk.length) {
    fs.closeSync(input);
    fs.rmSync(path, { force: true });
    process.exit(2);
  }
  for (let i = 0; i < got; i += 1) {
    acc = (acc + buf[i]) & 1048575;
  }
  total += got;
}
fs.closeSync(input);
fs.rmSync(path, { force: true });
if (total !== expected || acc === -1) {
  process.exit(3);
}

File: benchmarks/rust/file_io.rs
use std::fs;
use std::io::{Read, Write};

fn main() {
    let path = "tn_benchmark_file_io.tmp";
    let chunk = b"TezzNativeBenchmarkBlock";
    let reps = 12_000usize;
    let expected = chunk.len() * reps;

    let mut out = fs::File::create(path).unwrap_or_else(|_| std::process::exit(1));
    for _ in 0..reps {
        if out.write_all(chunk).is_err() {
            let _ = fs::remove_file(path);
            std::process::exit(2);
        }
    }
    drop(out);

    let mut input = fs::File::open(path).unwrap_or_else(|_| {
        let _ = fs::remove_file(path);
        std::process::exit(3);
    });
    let mut data = Vec::new();
    if input.read_to_end(&mut data).is_err() {
        let _ = fs::remove_file(path);
        std::process::exit(4);
    }
    let _ = fs::remove_file(path);
    if data.len() != expected {
        std::process::exit(5);
    }
    let mut acc: i32 = 0;
    for b in data {
        acc = (acc + (b as i32)) & 1_048_575;
    }
    if acc == -1 {
        std::process::exit(6);
    }
}
